lib/AST/Decl.cpp

```diff
@@ -419,6 +413,13 @@ void Decl::forEachAttachedMacro(MacroRole role,
   }
 }
 
+MacroDecl *Decl::getResolvedMacro(CustomAttr *customAttr) const {
```

Comment:
I suppose you can still mess this up by calling it on the wrong `Decl`. Perhaps there should be an assertion that `customAttr` appears in the decl's attribute list...

Reply:
That seems unlikely enough that I'm not worried about it. Some day I do want to get a ConcreteDeclRef out of this query so we have the substitution map for generic macros.